tic-tac-toe/src/tic_tac_toe/validating_prompt.clj

```diff
@@ -0,0 +1,51 @@
+(ns tic-tac-toe.validating-prompt
+  (:require [tic-tac-toe.writer :as writer]
+            [tic-tac-toe.reader :as reader]
+            [tic-tac-toe.board :as board]
+            ))
+
+(defn- zero-based [index]
+  (- index 1))
+
+(defn- vacant-space[index spaces-on-board]
+  (boolean (some #(= index %) spaces-on-board))
+  )
+
+(defn- show-invalid-message [error-msg board]
+  (writer/display board)
+  (error-msg))
+
+(defn- validate-input-is-within-range [provided-index board]
+  (if (vacant-space provided-index (board/indicies-of-free-spaces board))
+    true
+    (do
+      (show-invalid-message writer/invalid-space-message board)
+      false
+      ))
+  )
+
+(defn- to-number [input]
+  (Integer/parseInt input))
+
+(defn- validate-input-is-numeric [input board]
+  (try
+    (to-number input)
+    true
+    (catch Exception e
+      (show-invalid-message writer/not-numeric-message board)
+      false
+      )
+    ))
+
+(defn valid-next-move[board]
+  (writer/prompt-for-next-move)
+
+  (let [input (reader/read-input)]
+
+    (if (and (validate-input-is-numeric input board)
+             (validate-input-is-within-range (zero-based (to-number input)) board))
+      (zero-based (to-number input))
+      (valid-next-move board)
```

Comment:
I did refactor this differently, seen on a branch [here (line 44)](https://github.com/gemcfadyen/Apprenticeship-ClojureTicTacToe/blob/refactorings/tic-tac-toe/src/tic_tac_toe/validating_prompt.clj) However I thought the code looked showed duplication, was not concise, more complex to get the gist of what the function does.

I decided to revert back to what I have here, but am aware that the `validate-input-is....` functions display an error message and the board, as well as checking whether the input is valid,  which may not be obvious to a new developer to this project. 

As it's all about trade offs, I generally preferred this solution, and blogged about the journey [here under the heading pomodoro 5-8](http://gemcfadyen.github.io/georginam.com/apprenticeship/2016/03/15/apprenticeship-day-109.html)

Will be good to discuss further in the next IPM.